aws_xray_sdk/core/recorder.py

```diff
@@ -243,7 +243,8 @@ def begin_segment(self, name=None, traceid=None,
             self._populate_runtime_context(segment, decision)
 
         self.context.put_segment(segment)
-        return segment
+        current_segment = self.get_trace_entity()
+        return current_segment
```

Comment:
What is the intention here?

Reply:
In the case when the context modifies the segment under the hood and returns a (potentially) new entity, this ensures that if the returned entity is mutated at the higher levels, it would propagate down to the context.